comp

```diff
@@ -1,4 +1,27 @@
 #!/bin/sh
+{
+	for i in * /bin/* /usr/bin/* /usr/local/bin/* ~/bin/* ~/.local/bin/*
+	do
+		if sed 1q "$i" | grep '^#![ 	]*/bin/sh'
+		then
+			if ! grep '^# mark$' "$i"
```

Comment:
This is a wildly unnecessary use of grep(1). Instead, and much more efficiently, make use of _glob pattern matching_ within a _case_ statement:

```sh
case $i in
    \#\ mark)
        COMMANDS ;;
esac
```

Reply:
Are you a troll or can you just not read shell scripts? `$i` is the name of a file.

Reply:
This is GitHub, not Reddit. You needn't resort to such petty acts as name-calling.

Reply:
Okay, I am sorry, please disregard the name-calling question.